llvm/include/llvm/ObjectYAML/DXContainerYAML.h

```diff
@@ -74,16 +74,52 @@ struct ShaderHash {
 };
 
 #define ROOT_ELEMENT_FLAG(Num, Val) bool Val = false;
+
+struct RootConstantsYaml {
+  uint32_t ShaderRegister;
+  uint32_t RegisterSpace;
+  uint32_t Num32BitValues;
+};
+
+struct RootParameterYamlDesc {
+  dxbc::RootParameterType Type;
+  dxbc::ShaderVisibility Visibility;
+  uint32_t Offset;
+
+  RootParameterYamlDesc() = default;
+  RootParameterYamlDesc(const object::DirectX::RootParameter &Parameter) {
+
+    Type = Parameter.Header.ParameterType;
+    Visibility = Parameter.Header.ShaderVisibility;
+    Offset = Parameter.Header.ParameterOffset;
+    switch (Parameter.Header.ParameterType) {
+
+    case dxbc::RootParameterType::Constants32Bit: {
+      Constants.Num32BitValues = Parameter.Constants.Num32BitValues;
+      Constants.RegisterSpace = Parameter.Constants.RegisterSpace;
+      Constants.ShaderRegister = Parameter.Constants.ShaderRegister;
+    } break;
```

Comment:
A couple of odd style choices here:
- We don't need braces for this `case`
- If we did, the `break` should go inside the block.